provider/provider.go

```diff
@@ -213,25 +232,75 @@ func New(opts ...Option) (*SweepingProvider, error) {
 		connectivity.WithOfflineCheckInterval(cfg.connectivityCheckOfflineInterval),
 	)
 	if err != nil {
+		cleanup(prov.cleanupFuncs)
 		return nil, err
 	}
+	prov.cleanupFuncs = append(prov.cleanupFuncs, prov.connectivity.Close)
 
 	go prov.run()
 
 	return prov, nil
 }
 
+const retryInterval = 5 * time.Minute
```

Comment:
Generally, consts are better placed near top of file, It makes it easier to find all the things that control behavior or represent specific values.

google says:
In Go, it is a common and idiomatic practice to declare constants at the top of a file, typically after imports and before variables or function definitions.